core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -501,14 +501,18 @@ private[spark] class BlockManager(
       case Some(info) =>
         val level = info.level
         logDebug(s"Level for block $blockId is $level")
+        val taskAttemptId = Option(TaskContext.get()).map(_.taskAttemptId())
+          .getOrElse(BlockInfo.NON_TASK_WRITER)
```

Comment:
I think we can leave out the `.getOrElse` here and just pass in the `Option` itself into `releaseLock`. This helps to avoid exposure of `BlockInfo.NON_TASK_WRITER` here. Not a huge deal but just a minor nit.